drivers/adc/adaq8092/adaq8092.c

```diff
@@ -0,0 +1,683 @@
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <stdlib.h>
+#include <errno.h>
+#include "adaq8092.h"
+#include "no-os/delay.h"
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+/******************************************************************************/
+
+/**
+ * @brief Read device register.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - The data read from the register.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int adaq8092_read(struct adaq8092_dev *dev, uint8_t reg_addr, uint8_t *reg_data)
+{
+	int ret;
+	uint8_t buff[2] = {0};
+
+	buff[0] = ADAQ8092_SPI_READ | reg_addr;
+
+	ret = spi_write_and_read(dev->spi_desc, buff, 2);
+	if (ret)
+		return ret;
+
+	*reg_data = buff[1];
+
+	return 0;
+}
+
+/**
+ * @brief Write device register.
+ * @param dev- The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - The data to be written.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int adaq8092_write(struct adaq8092_dev *dev, uint8_t reg_addr, uint8_t reg_data)
+{
+	uint8_t buff[2] = {0};
+
+	buff[0] = reg_addr;
+	buff[1] = reg_data;
+
+	return spi_write_and_read(dev->spi_desc, buff, 2);
+}
+
+/**
+ * @brief Update specific register bits.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param mask - Specific bits mask.
+ * @param reg_data - The data to be written.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int adaq8092_update_bits(struct adaq8092_dev *dev, uint8_t reg_addr,
+			 uint8_t mask, uint8_t reg_data)
+{
+	int ret;
+	uint8_t data;
+
+	ret = adaq8092_read(dev, reg_addr, &data);
+	if (ret)
+		return ret;
+
+	data &= ~mask;
+	data |= reg_data;
+
+	return adaq8092_write(dev, reg_addr, data);
+}
+
+/**
+ * @brief Initialize the device.
+ * @param device - The device structure.
+ * @param init_param - The structure that contains the device initial
+ * 		       parameters.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int adaq8092_init(struct adaq8092_dev **device,
+		  struct adaq8092_init_param init_param)
```

Comment:
nitpick: we usually pass a pointer here

Reply:
ACK

Reply:
Actually after doing a search in the no-OS repository, I noticed that most of the init/setup functions for the drivers are not passing a pointer for the init_param :) (https://github.com/analogdevicesinc/no-OS/search?p=1&q=%2Ainit_param)